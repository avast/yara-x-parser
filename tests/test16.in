rule test
{
	strings:
		$a = { F4 [0-3] A4 [4] ( 62 (45 54 | 48) | 56 | 45 ?? 67 ) ~45 }
	condition:
		$a
}
