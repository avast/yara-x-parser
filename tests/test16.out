SOURCE_FILE@0..107
  RULE@0..106
    RULE_KW@0..4 "rule"
    WHITESPACE@4..5 " "
    IDENTIFIER@5..9 "test"
    WHITESPACE@9..10 "\n"
    BLOCK_EXPR@10..106
      L_BRACE@10..11 "{"
      WHITESPACE@11..13 "\n\t"
      STRINGS@13..87
        STRINGS_KW@13..20 "strings"
        COLON@20..21 ":"
        WHITESPACE@21..24 "\n\t\t"
        VARIABLE_STMT@24..87
          VARIABLE@24..26 "$a"
          WHITESPACE@26..27 " "
          ASSIGN@27..28 "="
          WHITESPACE@28..29 " "
          PATTERN@29..87
            HEX_PATTERN@29..87
              L_BRACE@29..30 "{"
              WHITESPACE@30..31 " "
              HEX_TOKEN@31..85
                HEX_BYTE@31..33
                  HEX_LIT@31..33 "F4"
                WHITESPACE@33..34 " "
                HEX_TOKEN_TAIL@34..42
                  HEX_JUMP@34..39
                    L_BRACKET@34..35 "["
                    LITERAL@35..36
                      INT_LIT@35..36 "0"
                    HYPHEN@36..37 "-"
                    LITERAL@37..38
                      INT_LIT@37..38 "3"
                    R_BRACKET@38..39 "]"
                  WHITESPACE@39..40 " "
                  HEX_BYTE@40..42
                    HEX_LIT@40..42 "A4"
                WHITESPACE@42..43 " "
                HEX_TOKEN_TAIL@43..45
                  HEX_BYTE@43..45
                    HEX_LIT@43..45 "B4"
                WHITESPACE@45..46 " "
                HEX_TOKEN_TAIL@46..81
                  HEX_ALTERNATIVE@46..81
                    L_PAREN@46..47 "("
                    WHITESPACE@47..48 " "
                    HEX_TOKEN@48..63
                      HEX_BYTE@48..50
                        HEX_LIT@48..50 "62"
                      WHITESPACE@50..51 " "
                      HEX_TOKEN_TAIL@51..63
                        HEX_ALTERNATIVE@51..63
                          L_PAREN@51..52 "("
                          HEX_TOKEN@52..57
                            HEX_BYTE@52..54
                              HEX_LIT@52..54 "45"
                            WHITESPACE@54..55 " "
                            HEX_TOKEN_TAIL@55..57
                              HEX_BYTE@55..57
                                HEX_LIT@55..57 "54"
                          WHITESPACE@57..58 " "
                          HEX_PIPE@58..62
                            PIPE@58..59 "|"
                            WHITESPACE@59..60 " "
                            HEX_TOKEN@60..62
                              HEX_BYTE@60..62
                                HEX_LIT@60..62 "48"
                          R_PAREN@62..63 ")"
                    WHITESPACE@63..64 " "
                    HEX_PIPE@64..68
                      PIPE@64..65 "|"
                      WHITESPACE@65..66 " "
                      HEX_TOKEN@66..68
                        HEX_BYTE@66..68
                          HEX_LIT@66..68 "56"
                    WHITESPACE@68..69 " "
                    HEX_PIPE@69..79
                      PIPE@69..70 "|"
                      WHITESPACE@70..71 " "
                      HEX_TOKEN@71..79
                        HEX_BYTE@71..73
                          HEX_LIT@71..73 "45"
                        WHITESPACE@73..74 " "
                        HEX_TOKEN_TAIL@74..76
                          HEX_BYTE@74..76
                            HEX_LIT@74..76 "??"
                        WHITESPACE@76..77 " "
                        HEX_TOKEN_TAIL@77..79
                          HEX_BYTE@77..79
                            HEX_LIT@77..79 "67"
                    WHITESPACE@79..80 " "
                    R_PAREN@80..81 ")"
                WHITESPACE@81..82 " "
                HEX_TOKEN_TAIL@82..85
                  HEX_BYTE@82..85
                    HEX_LIT@82..85 "~45"
              WHITESPACE@85..86 " "
              R_BRACE@86..87 "}"
      WHITESPACE@87..89 "\n\t"
      CONDITION@89..104
        CONDITION_KW@89..98 "condition"
        COLON@98..99 ":"
        WHITESPACE@99..102 "\n\t\t"
        EXPRESSION_STMT@102..104
          LITERAL@102..104
            VARIABLE@102..104 "$a"
      WHITESPACE@104..105 "\n"
      R_BRACE@105..106 "}"
  WHITESPACE@106..107 "\n"
